scripts/calendar/islamic_generator.py

```diff
@@ -0,0 +1,104 @@
+#!/usr/bin/env python3
+
+#  python-holidays
+#  ---------------
+#  A fast, efficient Python library for generating country, province and state
+#  specific sets of holidays on the fly. It aims to make determining whether a
+#  specific date is a holiday as fast and flexible as possible.
+#
+#  Authors: dr-prodigy <dr.prodigy.github@gmail.com> (c) 2017-2023
+#           ryanss <ryanssdev@icloud.com> (c) 2014-2017
+#  Website: https://github.com/dr-prodigy/python-holidays
+#  License: MIT (see LICENSE file)
+
+from pathlib import Path
+
+from hijridate import convert
```

Comment:
Should we create something like requirements/external.txt for this script dependencies?